src/xact.cc

```diff
@@ -211,13 +224,22 @@ bool xact_base_t::finalize()
   // If there is only one post, balance against the default account if one has
   // been set.
 
-  if (journal && journal->bucket && posts.size() == 1 && ! balance.is_null()) {
-    null_post = new post_t(journal->bucket, ITEM_INFERRED);
-    null_post->_state = (*posts.begin())->_state;
-    add_post(null_post);
+  if (journal && journal->bucket && posts.size() == 1) {
+    if (! balance_real.is_null()) {
+      null_post_real = new post_t(journal->bucket, ITEM_INFERRED);
+      null_post_real->_state = (*posts.begin())->_state;
+      add_post(null_post_real);
+    }
+    // TODO: does this need to set POST_VIRTUAL | POST_MUST_BALANCE ?
+    else if (! balance_virtual.is_null()) {
```

Comment:
Apparently there's some way to say "if an xact has only one posting, here's a default account to balance against".

If an xact has exactly one real and one balanced-virtual posting, right now I'm taking that to be an error. We could instead choose to balance both against the default account. But currently,

```
2024/08/27 Example
    Income        $1.00
    (Expenses)    $-1.00
```

doesn't balance against the default, so I wouldn't expect it to with `[]` either.